### PR TITLE
General components

### PR DESCRIPTION
**Pull request by tahmasious**
I've implemented main components (those which are used many times).
check the trello board out form more information : 
[boards link](https://trello.com/b/58litasE/kanban-template)

### CODE REVIEW

File: MyMovie/.env [hunk 1]
```diff
@@ -0,0 +1 @@
+VITE_API_READ_ACCESS_TOKEN=eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiI1ZTFlZGU1ZWZmZDI3MGY5YzA2ZGZjYmNiYzA5NTFjZSIsInN1YiI6IjY0NjY1MDZlYzM1MTRjMDExZGM5YzAyNyIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.SGUUPrpYPuP0z5LRA3_1xJOG4rxo1yOKa4d8x-6nu68
```
Comment (amRashidi) [inline MyMovie/.env]: add .env file to git ignore (use .env.example)
(VITE_API_READ_ACCESS_TOKEN=your token!!!)

File: MyMovie/src/App.vue [hunk 2]
```diff
@@ -1,9 +1,34 @@
 <template>
-  <TrendingMovieExpandCard title="The blade runner" imdbRate="8.3" publishYear="2016" backgroundImage="../src/assets/blade-runner.jpg" />
+  <div class="bg-[#19191E]">
+    <MainLayOut></MainLayOut>
+  </div>
+  <!-- <SideBarMoiveCard title="sag" coverImage="../src/assets/TLOU.jpg" :genres="['hello', 'pass']"></SideBarMoiveCard> -->
```
Comment (amRashidi) [inline MyMovie/src/App.vue]: remove comments

File: MyMovie/src/App.vue [hunk 3]
```diff
@@ -1,9 +1,34 @@
...
+import MainMovieCardVue from './components/general/MainMovieCard.vue';
+import RowOfMainMovieCard from './components/general/RowOfMainMovieCard.vue'
+import MainRightSideBar from './components/general/MainRightSideBar.vue';
+
+import {client} from './utils'
+import { ref } from 'vue';
+import MainLeftSideBar from './components/general/MainLeftSideBar.vue';
+import MainLayOut from './components/general/MainLayOut.vue';
+const objs = ref('');
+const res = client('https://api.themoviedb.org/3/movie/popular?language=en-US&page=1');
```
Comment (amRashidi) [inline MyMovie/src/App.vue]: use base url

File: MyMovie/src/components/general/AddToWatchListBtn.vue [hunk 4]
```diff
@@ -1,26 +1,26 @@
...
-    })
-</script>
+<template>
+    <a
+    href="#"
+    class="text-white text-lg bg-[#ffffff5e] rounded-2xl"
+    :style="{paddingTop : py, paddingBottom: py, paddingRight: px, paddingLeft: px}"
+    >
+    <img
+        src="../../assets/plus.png"
```
Comment (amRashidi) [inline MyMovie/src/components/general/AddToWatchListBtn.vue]: globalize url (@/)

File: MyMovie/src/components/general/AddToWatchListBtn.vue [hunk 5]
```diff
@@ -1,26 +1,26 @@
...
-            type: String,
-            default : '10px'
-        }
-    })
-</script>
+<template>
+    <a
+    href="#"
+    class="text-white text-lg bg-[#ffffff5e] rounded-2xl"
+    :style="{paddingTop : py, paddingBottom: py, paddingRight: px, paddingLeft: px}"
```
Comment (amRashidi) [inline MyMovie/src/components/general/AddToWatchListBtn.vue]: why??

File: MyMovie/src/components/general/ImbdContainer.vue [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    },
+    'textColor' : {
+        default : 'black'
+    }
+})
+</script>
+
+<style scoped>
+.imdb {
+  background-color: rgba(255, 255, 255, 0.1);
```
Comment (amRashidi) [inline MyMovie/src/components/general/ImbdContainer.vue]: bg-white background-opacity-10

File: MyMovie/src/components/general/LogoContainer.vue [hunk 7]
```diff
@@ -0,0 +1,27 @@
+<template>
+    <!-- logo -->
+    <div class="logo-container mb-4">
+        <img src="../../assets/logo.png" alt="mymovie logo" class="logo" />
```
Comment (amRashidi) [inline MyMovie/src/components/general/LogoContainer.vue]: address

File: MyMovie/src/components/general/LogoContainer.vue [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+        <h1>My movie<strong class="text-[35px] text-primary">.</strong></h1>
+    </div>
+</template>
+
+<script setup>
+
+</script>
+
+<style scoped>
+.logo {
```
Comment (amRashidi) [inline MyMovie/src/components/general/LogoContainer.vue]: tailwind class

File: MyMovie/src/components/general/LogoContainer.vue [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  width: 80px;
+  height: 80px;
+  display: table;
+  margin: auto;
+}
+.logo-container h1 {
```
Comment (amRashidi) [inline MyMovie/src/components/general/LogoContainer.vue]: same

File: MyMovie/src/components/general/MainLeftSideBar.vue [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+          class="w-7 h-7 rotate-45"
+          alt="close-btn"
+        />
+      </div>
+      <div class="sticky top-0">
+        <!-- logo -->
+        <logo-container />
+        <!-- main links -->
+        <section class="text-sm pb-3 pt-3">
+          <h2 class="nav-section-header">Personal</h2>
```
Comment (amRashidi) [inline MyMovie/src/components/general/MainLeftSideBar.vue]: do this by v-for

File: MyMovie/src/components/general/MainLeftSideBar.vue [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+          >
+        </div>
+      </div>
+    </nav>
+</template>
+
+<script setup>
+import LogoContainer from "./LogoContainer.vue";
+import navBarItem from './NavBarItem.vue';
+import tahmasiousPhoto from '../../assets/tahmasious.jpg'
```
Comment (amRashidi) [inline MyMovie/src/components/general/MainLeftSideBar.vue]: address

File: MyMovie/src/components/general/MainRightSideBar.vue [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+          <RedLinkBtn text="see all" href="#" />
+        </section>
+      </section>
+    </aside>
+</template>
+
+
+<script setup>
+import { ref } from 'vue';
+import {client} from './../../utils.js'
```
Comment (amRashidi) [inline MyMovie/src/components/general/MainRightSideBar.vue]: address

File: MyMovie/src/components/general/MainRightSideBar.vue [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+
+const topRatedMovies = ref([]);
+const upcomingMoives = ref([]);
+
+const res = client('https://api.themoviedb.org/3/movie/top_rated');
+res.then(data => {
+  topRatedMovies.value = data.results.slice(0,3);
+})
+
+const secRes = client('https://api.themoviedb.org/3/movie/upcoming');
```
Comment (amRashidi) [inline MyMovie/src/components/general/MainRightSideBar.vue]: base url

File: MyMovie/src/components/general/MainRightSideBar.vue [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+import { ref } from 'vue';
+import {client} from './../../utils.js'
+import SideBarMoiveCard from './SideBarMoiveCard.vue';
+import RedLinkBtn from './RedLinkBtn.vue'
+
+
+const topRatedMovies = ref([]);
+const upcomingMoives = ref([]);
+
+const res = client('https://api.themoviedb.org/3/movie/top_rated');
```
Comment (amRashidi) [inline MyMovie/src/components/general/MainRightSideBar.vue]: base url

File: MyMovie/src/components/general/MainRightSideBar.vue [hunk 15]
```diff
@@ -0,0 +1,80 @@
...
+res.then(data => {
+  topRatedMovies.value = data.results.slice(0,3);
+})
+
+const secRes = client('https://api.themoviedb.org/3/movie/upcoming');
+secRes.then(data => {
+  upcomingMoives.value = data.results.slice(0,3);
```
Comment (amRashidi) [inline MyMovie/src/components/general/MainRightSideBar.vue]: what if data length is lower than 3?

File: MyMovie/src/components/general/MainRightSideBar.vue [hunk 16]
```diff
@@ -0,0 +1,80 @@
...
+})
+
+</script>
+
+<style scoped>
+aside {
```
Comment (amRashidi) [inline MyMovie/src/components/general/MainRightSideBar.vue]: as i see this style is has repeated all over the sections.
please create a custom global class for this

File: MyMovie/src/components/general/NavBarItem.vue [hunk 17]
```diff
@@ -0,0 +1,40 @@
...
+  </a>
+</template>
+
+<script setup>
+  import { defineProps, ref } from 'vue';
+
+  const props = defineProps(['title', 'isActive', 'href']);
+  const importPath = `../../assets/${props.title}.png`;
+  const importPathActive = `../../assets/${props.title}-active.png`;
+  console.log(importPathActive);
```
Comment (amRashidi) [inline MyMovie/src/components/general/NavBarItem.vue]: remove log

File: MyMovie/src/components/general/NavBarItem.vue [hunk 18]
```diff
@@ -0,0 +1,40 @@
...
+    />
+    <span>{{ title }}</span>
+  </a>
+</template>
+
+<script setup>
+  import { defineProps, ref } from 'vue';
+
+  const props = defineProps(['title', 'isActive', 'href']);
+  const importPath = `../../assets/${props.title}.png`;
```
Comment (amRashidi) [inline MyMovie/src/components/general/NavBarItem.vue:16]: address

File: MyMovie/src/components/general/NavBarItem.vue [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+  const importPathActive = `../../assets/${props.title}-active.png`;
```
Comment (amRashidi) [inline MyMovie/src/components/general/NavBarItem.vue:17]: address

File: MyMovie/src/components/general/SideBarMoiveCard.vue [hunk 20]
```diff
@@ -0,0 +1,45 @@
+<template>
+    <div class="movie-container flex justify-start gap-2 my-2">
+        <div class="movie-poster-container">
+        <img
+            :src="`${API_IMAGE_BASE_URL}/w154${coverImagePath}`"
+            alt=""
```
Comment (amRashidi) [inline MyMovie/src/components/general/SideBarMoiveCard.vue]: fill wth title

File: MyMovie/src/utils.js [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+    if(!response.ok){
+        throw data;
+    }
+    return data;
+}
+
+
+export async function getGenreNamesByIDs(ids) {
+    let genreNames = []
+    const res = await client('https://api.themoviedb.org/3/genre/movie/list');
```
Comment (amRashidi) [inline MyMovie/src/utils.js]: base url

**Review by amRashidi — CHANGES_REQUESTED** [review]

File: MyMovie/src/components/general/AddToWatchListBtn.vue (hunk 5 above)
Comment (tahmasious) [reply MyMovie/src/components/general/AddToWatchListBtn.vue]: @amRashidi 
I didn't understand the problem

**Comment by tahmasious** [comment]
Merge main component branch to main

**Pull request merged**